xmppserver/src/main/java/org/jivesoftware/openfire/http/HttpSession.java

```diff
@@ -869,17 +870,15 @@ else if(connection.isPoll()) {
                 }
             }
             lastPoll = time;
-            if (Log.isDebugEnabled()) {
-                Log.debug("Updated session " + getStreamID() +
-                        " lastPoll to " + lastPoll +
-                        " with rid " + connection.getRequestId() +
-                        " lastResponseEmpty = " + lastResponseEmpty  +
-                        " overactivity = " + overactivity +
-                        " deltaFromLastPoll = " + deltaFromLastPoll +
-                        " isPollingSession() = " + localIsPollingSession +
-                        " maxRequests = " + maxRequests +
-                        " pendingConnections = " + pendingConnections);
-            }
+            Log.debug("Updated session " + getStreamID() +
+                    " lastPoll to " + lastPoll +
+                    " with rid " + connection.getRequestId() +
+                    " lastResponseEmpty = " + lastResponseEmpty  +
+                    " overactivity = " + overactivity +
+                    " deltaFromLastPoll = " + deltaFromLastPoll +
+                    " isPollingSession() = " + localIsPollingSession +
+                    " maxRequests = " + maxRequests +
+                    " pendingConnections = " + pendingConnections);
```

Comment:
I think the isDebugEnabled was to avoid the string being built, which is still happening. The other lines change this to {} constructs, which won't build the string unless it'll be logged.

Reply:
That's easily fixed. I've added a commit that does just this to the PR.